x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/cohere/rerank/CohereRerankServiceSettings.java

```diff
@@ -7,43 +7,125 @@
-import org.elasticsearch.xpack.inference.services.cohere.CohereServiceSettings;
+import org.elasticsearch.xpack.inference.services.cohere.CohereRateLimitServiceSettings;
+import org.elasticsearch.xpack.inference.services.cohere.CohereService;
 import org.elasticsearch.xpack.inference.services.settings.FilteredXContentObject;
+import org.elasticsearch.xpack.inference.services.settings.RateLimitSettings;
 
 import java.io.IOException;
+import java.net.URI;
 import java.util.Map;
 import java.util.Objects;
 
-public class CohereRerankServiceSettings extends FilteredXContentObject implements ServiceSettings {
+import static org.elasticsearch.xpack.inference.services.ServiceFields.DIMENSIONS;
+import static org.elasticsearch.xpack.inference.services.ServiceFields.MAX_INPUT_TOKENS;
+import static org.elasticsearch.xpack.inference.services.ServiceFields.MODEL_ID;
+import static org.elasticsearch.xpack.inference.services.ServiceFields.URL;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.convertToUri;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.createOptionalUri;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalString;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractSimilarity;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.removeAsType;
+import static org.elasticsearch.xpack.inference.services.cohere.CohereServiceSettings.DEFAULT_RATE_LIMIT_SETTINGS;
+import static org.elasticsearch.xpack.inference.services.cohere.CohereServiceSettings.OLD_MODEL_ID_FIELD;
+
+public class CohereRerankServiceSettings extends FilteredXContentObject implements ServiceSettings, CohereRateLimitServiceSettings {
     public static final String NAME = "cohere_rerank_service_settings";
 
-    public static CohereRerankServiceSettings fromMap(Map<String, Object> map, ConfigurationParseContext parseContext) {
+    private static final Logger logger = LogManager.getLogger(CohereRerankServiceSettings.class);
+
+    public static CohereRerankServiceSettings fromMap(Map<String, Object> map, ConfigurationParseContext context) {
         ValidationException validationException = new ValidationException();
-        var commonServiceSettings = CohereServiceSettings.fromMap(map, parseContext);
+
+        String url = extractOptionalString(map, URL, ModelConfigurations.SERVICE_SETTINGS, validationException);
+
+        // We need to extract/remove those fields to avoid unknown service settings errors
+        extractSimilarity(map, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        removeAsType(map, DIMENSIONS, Integer.class);
+        removeAsType(map, MAX_INPUT_TOKENS, Integer.class);
+
+        URI uri = convertToUri(url, URL, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        String oldModelId = extractOptionalString(map, OLD_MODEL_ID_FIELD, ModelConfigurations.SERVICE_SETTINGS, validationException);
```

Comment:
Hmm, we added the logic to switch from the old model id to the new one before the cohere rerank integration:

[Rerank integration](https://github.com/elastic/elasticsearch/commit/f49ead7446e722c5cf6d6268e3a043dd1f818fce)

[Transition from old model id](https://github.com/elastic/elasticsearch/commit/f1b4878c72231bb108347a5c31247bbc68002426)

How do we feel about removing the functionality to check for the old field? @davidkyle @maxhniebergall 

I guess it's possible that users were using the deprecated `model` field so if they attempted to create a new rerank inference endpoint it'd start failing. The field already being persisted as `model_id` in elasticsearch and would continue to do that so I don't think we'd have a backwards compatibility issue here 🤔

Reply:
Imo, its best to deprecate these old field strings asap. we definitely don't want to have this kind of bwc support when we move to GA

Reply:
Adjusted with [Remove old model id check](https://github.com/elastic/elasticsearch/pull/110427/commits/c5f3e7d74fab067a423098dfe011047991a3be6b)